azure/pf-azure-context/functions/public/set-pfazurecontext.ps1

```diff
@@ -1,13 +1,40 @@
-$global:_currentPfAzureContext
+$global:_PfAzureContext
 Function Set-PfAzureContext 
 {
     Param (
-        [Parameter(Mandatory = $true)] 
+        [Parameter(Mandatory = $true, ParameterSetName="All")] 
+        [Parameter(Mandatory = $true, ParameterSetName="AzureInfo")] 
         [string] $SubscriptionName,
-        [Parameter(Mandatory = $true)] 
-        [string] $ResourceGroupName
+        [Parameter(Mandatory = $true, ParameterSetName="All")] 
+        [Parameter(Mandatory = $true, ParameterSetName="AzureInfo")] 
+        [string] $ResourceGroupName,
+        [Parameter(Mandatory = $true, ParameterSetName="All")] 
+        [Parameter(Mandatory = $true, ParameterSetName="OrgInfo")] 
+        [string] $CompanyAbbreviation,
+        [Parameter(Mandatory = $true, ParameterSetName="All")] 
+        [Parameter(Mandatory = $true, ParameterSetName="OrgInfo")] 
+        [string] $GroupAbbreviation,
+        [Parameter(Mandatory = $true, ParameterSetName="All")] 
+        [Parameter(Mandatory = $true, ParameterSetName="OrgInfo")] 
+        [string] $TeamAbbreviation,
+        [Parameter(Mandatory = $true, ParameterSetName="All")] 
+        [Parameter(Mandatory = $true, ParameterSetName="OrgInfo")] 
+        [string] $EnvironmentLetter,
+        [Parameter(Mandatory = $true, ParameterSetName="All")] 
+        [Parameter(Mandatory = $true, ParameterSetName="AzureInfo")] 
+        [Parameter(Mandatory = $true, ParameterSetName="OrgInfo")] 
+        [string] $Region
```

Comment:
interesting that region crosses over parm groups.  as i was looking at context constructors above was thinking maybe these could be grouped and set together in context based on 'group' so that is reused across constructors (for the forget to change in one place issue), and region was seeming to be in the 'azure info' category in my mind.

Reply:
So this is the only way I know how to overload a function.  So you have 3 options when you create a new az context.  You can set all params, which RGName and SubName would override any default names created by the orginfo.  Or you can just supply orginfo and it will create naming standards.  Or you can just supply AzureInfo, and you then have to define the rgname and subname.